filecache/file_cache_path.py

```diff
@@ -1118,150 +1157,233 @@ def replace(self, target: str | FCPath) -> None:
+        working directory. No normalization or symlink resolution is performed.
 
-    def absolute(self) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+        Use resolve() to get the canonical path to a file.
+        """
+        if self.is_local():
+            return FCPath(self.get_local_path().absolute())
+        return self
 
     @classmethod
-    def cwd(cls) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def cwd(cls) -> FCPath:
+        """Return a new path pointing to the current working directory."""
+        return FCPath(Path.cwd())
 
-    def expanduser(self) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def expanduser(self) -> FCPath:
+        """Return a new path with expanded ~ and ~user constructs
+        (as returned by os.path.expanduser)
+        """
+        if self.is_local():
+            return FCPath(self.get_local_path().expanduser())
+        return self
 
     @classmethod
-    def home(cls) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def home(cls) -> FCPath:
+        """Return a new path pointing to the user's home directory (as
+        returned by os.path.expanduser('~')).
+        """
+        return FCPath(Path.home())
 
-    def readlink(self) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+    def readlink(self) -> FCPath:
+        """Return the path to which the symbolic link points."""
+        if self.is_local():
+            return FCPath(self.get_local_path().readlink())
+        return self     # no symlinks, so it points to itself
 
     def resolve(self,
-                strict: bool = False) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+                strict: bool = False) -> FCPath:
+        """Make the path absolute, resolving all symlinks on the way and also
+        normalizing it.
+        """
+        if self.is_local():
+            return FCPath(self.get_local_path().resolve(strict=strict))
+        return self
 
     def symlink_to(self,
-                   target: str,
+                   target: str | Path | FCPath,
                    target_is_directory: bool = False) -> None:
-        """Path function not supported by FCPath."""
-        raise NotImplementedError
+        """Make this path a symlink pointing to the target path.
+        Note the order of arguments (link, target) is the reverse of os.symlink.
+        """
+        if self.is_local():
+            target_fcpath = FCPath(target)
+            Path(self._path).symlink_to(target=target_fcpath.get_local_path(),
+                                        target_is_directory=target_is_directory)
+        raise NotImplementedError('FCPath.symlink_to')
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider restructuring symlink_to implementation flow</b></div>
<div id="fix">

The `raise NotImplementedError` statement after the `if` block means non-local paths will always raise an error, making the `if` block unreachable for those cases. Consider moving the error before the `if` block for non-local paths.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
        if not self.is_local():
            raise NotImplementedError('FCPath.symlink_to for non-local paths')
        target_fcpath = FCPath(target)
        Path(self._path).symlink_to(target=target_fcpath.get_local_path(),
                                    target_is_directory=target_is_directory)
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/SETI/rms-filecache/pull/31#issuecomment-2558269998>#e24627</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged